app/containers/gistEditorForm/index.js

```diff
@@ -84,8 +87,12 @@ class GistEditorFormImpl extends Component {
 const valideNotEmptyContent = value => value ? null : 'required'
 
 const validateFilename = value => {
-  if (!value) return 'required'
-  else if (!validFilename(value)) return 'invalid filename'
+  if (conf.get('editor').validateFilename) {
+    if (!value) return 'required'
```

Comment:
This check if the filename is empty otherwise shows "required". I don't think we should skip this check.